benchmark/paddle/QMIX/qmix_agent.py

```diff
@@ -0,0 +1,111 @@
+#   Copyright (c) 2021 PaddlePaddle Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import paddle
+import parl
+import numpy as np
+from utils import AvailableActionsSampler
+import os
+
+
+class QMixAgent(parl.Agent):
+    def __init__(self, algorithm, exploration_start, min_exploration,
+                 exploration_decay, update_target_interval):
+        self.alg = algorithm
+        self.global_step = 0
+        self.exploration = exploration_start
+        self.min_exploration = min_exploration
+        self.exploration_decay = exploration_decay
+        self.target_update_count = 0
+        self.update_target_interval = update_target_interval
+
+    def save(self, save_dir, agent_model_name, qmixer_model_name):
+        if not os.path.exists(save_dir):
+            os.mkdir(save_dir)
+        agent_model_path = os.path.join(save_dir, agent_model_name)
+        qmixer_model_path = os.path.join(save_dir, qmixer_model_name)
+        paddle.save(self.alg.agent_model.state_dict(), agent_model_path)
+        paddle.save(self.alg.qmixer_model.state_dict(), qmixer_model_path)
+        print('save model successfully!')
+
+    def restore(self, save_dir, agent_model_name, qmixer_model_name):
+        if not os.path.exists(save_dir):
+            os.mkdir(save_dir)
+        agent_model_path = os.path.join(save_dir, agent_model_name)
+        qmixer_model_path = os.path.join(save_dir, qmixer_model_name)
+        self.alg.agent_model.set_state_dict(paddle.load(agent_model_path))
+        self.alg.qmixer_model.set_state_dict(paddle.load(qmixer_model_path))
+        print('restore model successfully!')
+
+    def reset_agent(self, batch_size=1):
+        self.alg._init_hidden_states(batch_size)
+
+    def sample(self, obs, available_actions):
+        """ Args:
```

Comment:
Use same code style of comments, like the following examle:
```
        """ take greedy actions
        Args:
            obs:                (n_agents, obs_shape)
            available_actions:  (n_agents, n_actions)
        """
```

Reply:
fixed